src/ansys/mapdl/core/hpc/pyhps.py

```diff
@@ -0,0 +1,373 @@
+    if param_mappings is not None:
+        raise NotImplementedError("'param_mappings' is not implemented")
+    else:
+        param_mappings = []
+        logger.debug("Setting empty parameter mappings")
+    return project_api.create_parameter_mappings(param_mappings)
+
+
+def create_task(
+    project_api,
+    main_file,
+    file_input_ids,
+    file_output_ids,
+    num_cores,
+    memory,
+    disk_space,
+    exclusive,
+    max_execution_time,
+):
+
+    software = Software(name="Bash", version="0.1")  # Overwriting
+    execution_command = f"%executable% %file:{os.path.basename(main_file)}%"
+    logger.debug(f"Using executable: '{execution_command}'")
+
+    # Process step
+    task_def = TaskDefinition(
+        name="PyMAPDL_task",
+        software_requirements=[software],
+        execution_command=execution_command,
+        resource_requirements=ResourceRequirements(
+            num_cores=int(num_cores),
+            memory=int(memory) * 1024 * 1024,
+            disk_space=int(disk_space) * 1024 * 1024,
+            # distributed=True,
+            hpc_resources=HpcResources(exclusive=exclusive),
+        ),
+        max_execution_time=max_execution_time,
+        execution_level=0,
+        num_trials=1,
+        input_file_ids=list(file_input_ids.values()),
+        output_file_ids=list(file_output_ids.values()),
+    )
+    logger.debug(f"Task definition: {task_def}")
+
+    return project_api.create_task_definitions([task_def])[0]
+
+
+def create_job_definition(
+    project_api,
+    task_def,
+    input_params,
+    output_params,
+    param_mappings,
+):
+    job_def = JobDefinition(name="JobDefinition.1", active=True)
+    params = input_params + output_params
+
+    job_def.task_definition_ids = [task_def.id]
+    job_def.parameter_definition_ids = [pd.id for pd in params]
+    job_def.parameter_mapping_ids = [pm.id for pm in param_mappings]
+
+    logger.debug(f"Job definition: {job_def}")
+    job_def = project_api.create_job_definitions([job_def])[0]
+
+    # Refresh the parameters
+    params = project_api.get_parameter_definitions(id=job_def.parameter_definition_ids)
+    return job_def
+
+
+def create_jobs(project_api, job_def):
+    jobs = [
+        Job(name="Job", values={}, eval_status="pending", job_definition_id=job_def.id)
+    ]
+    logger.debug(f"jobs: {jobs}")
+    return project_api.create_jobs(jobs)
+
+
+def get_project_api(client, proj):
+    return ProjectApi(client, proj.id)
+
+
+def wait_for_completion(project_api, evaluated=True, failed=False, running=False):
+    eval_status = []
+
+    if evaluated:
+        eval_status.append("evaluated")
+
+    if failed:
+        eval_status.append("evaluated")
+
+    if running:
+        eval_status.append("running")
+
+    logger.debug(f"Waiting on project {proj.id} with criteria: {eval_status}")
+    while not project_api.get_jobs(eval_status=eval_status):
+        time.sleep(2)
+
+
+def create_tmp_file(file_name, content):
```

Comment:
use a preceding underscore since this is not meant to be used from outside pyhps